Tests/ToolBox/ForexVolume/ForexVolumeDownloaderTest.cs

```diff
@@ -0,0 +1,342 @@
+﻿using Ionic.Zip;
+using NodaTime;
+using NUnit.Framework;
+using QuantConnect.Data;
+using QuantConnect.Data.Custom;
+using QuantConnect.ToolBox;
+using QuantConnect.ToolBox.FxVolumeDownloader;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace QuantConnect.Tests.ToolBox.FxVolume
+{
+    [SetUpFixture]
+    public class SetUpClass
+    {
+        [SetUp]
+        public void SetUpTests()
+        {
+            Market.Add("FXCMForexVolume", identifier: 20);
+        }
+    }
+
+    [TestFixture]
+    public class ForexVolumeDownloaderTest
+    {
+        [SetUp]
+        public void SetUpTemporatyFolder()
+        {
+            var randomFolder = "ForexVolumeTesting_" + Guid.NewGuid().ToString("N").Substring(startIndex: 0, length: 6);
+            _dataDirectory = Path.Combine(Path.GetTempPath(), randomFolder);
+            _downloader = new ForexVolumeDownloader(_dataDirectory);
```

Comment:
Is there anyway to rethink these tests so we don't touch things we dont own? Maybe rethink your generator class methods so it outputs a string which is separately written to disk by your program. This way you can string compare the output of the generator & avoid testing writing to disk,